### PR TITLE
Postprocess: Cut long filenames to 255 chars max.

### PR DESCRIPTION
**Pull request by p0psicles**
- [ ] PR is based on the DEVELOP branch
- [ ] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [ ] Read the [contribution guide](https://github.com/pymedusa/Medusa/blob/master/.github/CONTRIBUTING.md)

Fix for #3699

### CODE REVIEW

**Comment by fernandog** [comment]
Will PP work with the cropped filename?

**Comment by p0psicles** [comment]
@fernandog i'm testing it in practice. So until now it seems to work fine. Had to take into account that it wil try to add `-thumb.jpg` to the basename.

**Comment by p0psicles** [comment]
Okay going a different route. When the resulting basename is > 245 chars, i'm removing the episode name, if it's part of the naming pattern.

File: medusa/tv/episode.py [hunk 1]
```diff
@@ -1702,12 +1702,20 @@ def _format_pattern(self, pattern=None, multi=None, anime_type=None):
                 # cur_name_group_result = cur_name_group.replace(ep_format, ep_string)
                 result_name = result_name.replace(cur_name_group, cur_name_group_result)
 
-        result_name = self.__format_string(result_name, replace_map)
+        parsed_result_name = self.__format_string(result_name, replace_map)
+
+        if len(parsed_result_name) > 244 and any(['%E.N' in result_name, '%EN' in result_name, '%E_N' in result_name]):
```
Comment (medariox) [inline medusa/tv/episode.py]: Why > 244? A little comment would be nice to not forget

File: medusa/tv/episode.py [hunk 2]
```diff
@@ -1702,12 +1702,20 @@ def _format_pattern(self, pattern=None, multi=None, anime_type=None):
...
+            for remove_pattern in ('%E.N', '%EN', '%E_N'):
+                result_name = result_name.replace(remove_pattern, '')
+            result_name = result_name.strip('- ')
```
Comment (medariox) [inline medusa/tv/episode.py]: Same with the strip('- ')

File: medusa/tv/episode.py [hunk 3]
```diff
@@ -1702,12 +1702,25 @@ def _format_pattern(self, pattern=None, multi=None, anime_type=None):
...
+
+        # With the episode name filenames tend to grow very large. Worst case scenario we even need to add `-thumb.jpg`
+        # to the filename. To make sure we stay under the 255 character limit, we're working with 244 chars, taking into
+        # account the thumbnail.
+        if len(parsed_result_name) > 244 and any(['%E.N' in result_name, '%EN' in result_name, '%E_N' in result_name]):
+            for remove_pattern in ('%E.N', '%EN', '%E_N'):
+                result_name = result_name.replace(remove_pattern, '')
+            # The Episode name can be appended with a - or . in between. Therefor we're removing it.
+            # Creating a clean filename.
+            result_name = result_name.strip('-. ')
```
Comment (medariox) [inline medusa/tv/episode.py]: strip() only remove from the beginning or the end of the string. Is it the case here?

Comment (p0psicles) [reply medusa/tv/episode.py]: Yes. Maybe a user puts the episode name a the beginning. At least that was my reasoning for it.

**Comment by p0psicles** [comment]
https://github.com/SickRage/SickRage/issues/4285

**Pull request merged**